src/graphics.c

```diff
@@ -1156,10 +1160,15 @@ void insta_fadeout(void)
   else
     num_colors = PAL_SIZE;
 
+  memcpy(graphics.saved_intensity, graphics.current_intensity,
+   sizeof(Uint32) * num_colors);
+
   for(i = 0; i < num_colors; i++)
     set_color_intensity(i, 0);
 
+#ifndef __EMSCRIPTEN__
   delay(1);
+#endif
```

Comment:
This `delay` before `update_screen` thing seems really sketchy to me in general. If you think about how `dialog_fadeout` is used all this achieves is delaying 1ms more on the UI that was just drawn/delayed on. Similar situation with closing the editor and `LOAD_GAME`. Why?

Added in 2.80, same version this function for some reason _stopped_ saving the intensities. The delay might be safe to remove altogether.

Reply:
This is removed in `master`.

Reply:
Rebased and adjusted accordingly.